snuba/datasets/errors_replacer.py

```diff
@@ -148,89 +157,161 @@ def get_count_query(self, table_name: str) -> Optional[str]:
+    needs_final_keys_and_type_keys = [
+        get_project_needs_final_key_and_type_key(project_id, state_name)
         for project_id in s_project_ids
     ]
-    for needs_final_key in needs_final_keys:
+
+    for needs_final_key, _ in needs_final_keys_and_type_keys:
         p.get(needs_final_key)
 
-    exclude_groups_keys = [
-        get_project_exclude_groups_key(project_id, state_name)
+    exclude_groups_keys_and_types = [
+        get_project_exclude_groups_key_and_type_key(project_id, state_name)
         for project_id in s_project_ids
     ]
-    for exclude_groups_key in exclude_groups_keys:
+
+    for exclude_groups_key, _ in exclude_groups_keys_and_types:
         p.zremrangebyscore(
             exclude_groups_key, float("-inf"), now - settings.REPLACER_KEY_TTL
         )
         p.zrevrangebyscore(
             exclude_groups_key, float("inf"), now - settings.REPLACER_KEY_TTL
         )
 
+    for _, needs_final_type_key in needs_final_keys_and_type_keys:
+        p.get(needs_final_type_key)
+
+    for _, type_key in exclude_groups_keys_and_types:
+        p.zremrangebyscore(type_key, float("-inf"), now - settings.REPLACER_KEY_TTL)
+        p.zrevrangebyscore(type_key, float("inf"), now - settings.REPLACER_KEY_TTL)
+
     results = p.execute()
 
-    needs_final = any(results[: len(s_project_ids)])
+    return process_exclude_groups_and_replacement_types_results(
+        results, len(s_project_ids)
+    )
+
+
+def process_exclude_groups_and_replacement_types_results(
+    results: List[Any], len_projects: int
+) -> Tuple[bool, Sequence[int], Set[str]]:
+    """
+    Helper function for `get_projects_query_flags`.
+    `results` is in the form:
+    [
+        needs_final...,
+        exclude_groups...,
+        project_replacement_types...,
+        groups_replacement_types...
+    ]
+    - `needs_final` slice is `len_projects` long
+    - `excludes_groups` slice is `len_projects * 2` long
+    - `project_replacement_types` slice is `len_projects` long
+    - `groups_replacement_types` slice is `len_projects * 2` long
+
+    The `len_projects * 2` long slices are in the form:
+    int, list, int, list, ...
+    Only the lists are necessary, the ints are scores from
+    the redis sorted set used to track when the items were zadded.
+    """
+
+    needs_final = any(results[:len_projects])
+    exclude_groups_results = results[len_projects : len_projects * 3]
+    projects_replacment_types_result = results[len_projects * 3 : len_projects * 4]
+    groups_replacement_types_result = results[len_projects * 4 :]
+
     exclude_groups = sorted(
-        {int(group_id) for group_id in sum(results[(len(s_project_ids) + 1) :: 2], [])}
+        {int(group_id) for group_id in sum(exclude_groups_results[1::2], [])}
     )
 
-    return (needs_final, exclude_groups)
+    project_replacement_types = {
+        replacement_type.decode("utf-8")
+        for replacement_type in projects_replacment_types_result
+        if replacement_type
```

Comment:
Project replacements are queried from redis using `get()` which will add a `None` to the results list if there is nothing at the key, hence `if replacement_type`. The loop below is looking through the results of `zrevrangebyscore()` which will just return an empty list instead since it's looking for a range and not a single result. `projects_replacment_types_result` is the flattened list which won't have any `None`'s so no need to check `if replacement_type`.